.github/workflows/approved-for-ci-run.yml

```diff
@@ -54,6 +54,7 @@ jobs:
       - uses: actions/checkout@v3
         with:
           ref: main
+          token: ${{ secrets.CI_ACCESS_TOKEN }}
```

Comment:
On the second thought, we'd better use the default `GITHUB_TOKEN` (which expires after min(job finish, 24h) ) and set proper [`permissions`](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token).
I will create a separate issue for that (it will require some testing).